docs/style-guide.md

```diff
@@ -0,0 +1,201 @@
+# ECMA-402 Style Guide
+
+This document is an evolving list of recommendations for the ECMA-402 specification.  Other specifications may refer to this document if desired.  Also see the [W3C TAG Design Principles](https://w3ctag.github.io/design-principles/#casing-rules) for additional guidance.
+
+This document contains background on how the style decisions were reached.  The actual recommendations are highlighted with a :star2: emoji.
+
+## Table of Contents
+
+- [Casing Conventions](#casing-conventions)
+    - [Identifiers defined by ECMA-402](#identifiers-defined-by-ecma-402)
+        - [Examples](#examples)
+        - [Alternative: Kebab case for all string enumerations](#alternative-kebab-case-for-all-string-enumerations)
+            - [Pros](#pros)
+            - [Cons](#cons)
+            - [Decision](#decision)
+        - [Alternative: Kebab case for new identifiers only](#alternative-kebab-case-for-new-identifiers-only)
+            - [Pros](#pros-1)
+            - [Cons](#cons-1)
+            - [Decision](#decision-1)
+        - [Alternative: Use kebab case but also accept camel case](#alternative-use-kebab-case-but-also-accept-camel-case)
+            - [Pros](#pros-2)
+            - [Cons](#cons-2)
+            - [Decision](#decision-2)
+    - [Identifiers defined outside ECMA-402](#identifiers-defined-outside-ecma-402)
+        - [Examples](#examples-1)
+        - [Alternative: Convert identifiers to camel case](#alternative-convert-identifiers-to-camel-case)
+            - [Pros](#pros-3)
+            - [Cons](#cons-3)
+            - [Decision](#decision-3)
+
+
+*Table of Contents generated using https://magnetikonline.github.io/markdown-toc-generate/*
```

Comment:
I'm fine adding this, but I wonder if we should move it to the end of the file.

Reply:
There isn't really anywhere to put it down there.  I like having it close to the actual ToC so it's easy to find how to update it.